lib/scholar/integrate/integrate.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Scholar.Integrate do
+  @moduledoc """
+  Module for numerical integration.
+  """
+
+  import Nx.Defn
+
+  @doc """
+  Integrate `y` along the given axis using the chosen rule.
+
+  ## Options
+
+  Options are the same as for the chosen rule. See the documentation for
+  specific implementations for more information.
+
+  ## Examples
+
+      iex> Scholar.Integrate.integrate(Nx.tensor([1, 2, 3]), :simpson, Nx.tensor([4, 5, 6]))
+      #Nx.Tensor<
+        f32
+        4.0
+      >
+
+      iex> Scholar.Integrate.integrate(Nx.tensor([[0, 1, 2], [3, 4, 5]]), :trapezoidal, Nx.tensor([[1, 2, 3], [1, 2, 3]]))
+      #Nx.Tensor<
+        f32[2]
+        [2.0, 8.0]
+      >
+
+      iex> Scholar.Integrate.integrate(Nx.tensor([[0, 1, 2], [3, 4, 5]]), :simpson, Nx.tensor([[1, 1, 1], [2, 2, 2]]), axis: 0)
+      #Nx.Tensor<
+        f32[3]
+        [1.5, 2.5, 3.5]
+      >
+  """
+  deftransform integrate(y, method, x, opts \\ []) do
```

Comment:
I would have the `method` as first argument, given it is the most important (and the most different argument).

Reply:
Having `y` as the first argument lends the function more easily to pipe-chains. I would have the method as the last mandatory argument to keep all data arguments together, though

Reply:
I would prefer `(method, y, x)` but `(y, x, method)` works for me too.

Reply:
In isolation `(method, y, x)` does look better, but it makes for clunky pipes with the need for `then` if you're piping `y` into `integrate`